### ISSUE TITLE
Has anyone created a script to pull US County level data directly from the Johns Hopkins Dashboard?

### ISSUE THREAD

**Issue by robertcox87** [CSSEGISandData/COVID-19#1788]

**Comment by J-Rojas** [comment]
Take a look at the web-data branch for the Dashboard data.

**Comment by filotti** [comment]
You don't need to pull the data from the dashboard, everything's available in this repo.

I made a node.js script to parse the daily csv files from csse_covid_19_data/csse_covid_19_daily_reports/, starting from 2020-03-22 there is county level data.

Here's the script so you can tweak it to fit your needs:

`const parse = require('csv-parse/lib/sync');
const fs = require('fs');
const path = require('path');
const source = 'csse_covid_19_daily_reports/';

const data = {};

const parseRow = (row, data) => {
    const FIPS = row.FIPS.padStart(5, '0');
    if (typeof data[FIPS] === 'undefined') {
        data[FIPS] = {
            lat: row.Lat,
            long: row.Long_,
            dates: {}
        }
    }
    const date = new Date(row.Last_Update);
    const dateString = date.getFullYear() + '-' + ('0' + (date.getMonth()+1)).slice(-2) + '-' + ('0' + date.getDate()).slice(-2);
    if (typeof data[FIPS]['dates'][dateString] === 'undefined') {
        data[FIPS]['dates'][dateString] = {
            confirmed: row.Confirmed
        }
    }
};

const parseCSV = (filename, data) => {
    console.log("Parsing ", filename);

    const rows = parse(fs.readFileSync( source + filename ), { columns: true , bom: true});

    for (let i = 0; i < rows.length; i++) {
        if (rows[i].Country_Region === 'US') {
            parseRow(rows[i], data);
        }
    }

};

const items = fs.readdirSync(source);

for (let i = 0; i < items.length; i++) {
    if (path.extname(items[i]) === '.csv') {
        parseCSV(items[i], data);
    }
}

fs.writeFileSync('data.json', JSON.stringify(data));

console.log('Done!');`

**Comment by robertcox87** [comment]
Thanks! I was looking for the dashboard because it seemed to be more current than the static .csv. Any idea if there is a love county feed available anywhere? Thanks again for the help!

Sent from Outlook Mobile<https://aka.ms/blhgte>

________________________________

**Comment by filotti** [comment]
You're right, I wish they would update this project more often. I don't think you can access the data directly, at least not without a significant amount of effort.

**Comment by J-Rojas** [comment]
They have automated scripts uploading the data continuously. If it's out of sync, it probably is only out of sync for a short period of time (maybe a few hours)

**Comment by J-Rojas** [comment]
The country data is in the web-data branch BTW. Just look at the cases.csv there.

**Comment by J-Rojas** [comment]
They just published a US county level time series report in the master branch as well.

**Comment by gohkokhan** [comment]
Yes, I've created a script to pull the data directly from the dashboard in **real-time** here https://github.com/gohkokhan/covid19_JHU_dashboard
I've been using it for a month for my interactive dashboard on Data Studio https://datastudio.google.com/reporting/f56febd8-5c42-4191-bcea-87a3396f4508

However, currently, I cannot get the data for county level, because I need JHU change the configuration on their feature server. https://services1.arcgis.com/0MSEUqKaxRlEPj5g/arcgis/rest/services/ncov_cases/FeatureServer/1
It's mentioned here: https://github.com/CSSEGISandData/COVID-19/issues/1250#issuecomment-606351561

**Comment by jjbenes** [comment]
county-level and state-level maps [here](http://first-principles.ai/covid-19/map.html) based [nyt](https://github.com/nytimes/covid-19-data) data... should probably switch to the new [file](https://github.com/CSSEGISandData/COVID-19/blob/master/csse_covid_19_data/csse_covid_19_time_series/time_series_covid19_confirmed_US.csv) from jhu

**Comment by robertcox87** [comment]
Thanks for the update!

**Comment by jjbenes** [comment]
Okay. I just switched from the NYT database to the JHU US-county CSV files. The JHU team did a nice job with all these files. It wasn't too painful to switch the front-end. I used to combine population data from the U.S. Census Bureau and the NYT CSV files to get the choropleths. Now I'm using only CSV files from JHU.

The map with four choropleths, two for the state level and two for the county level, is at the same [place](https://first-principles.ai/covid-19/map.html).

![Screen Shot 2020-04-02 at 1 52 35 AM](https://user-images.githubusercontent.com/62985029/78229608-18934b00-7485-11ea-8a2a-615d326f1075.png)
).

**Comment by J-Rojas** [comment]
Nice map! BTW where did you get the county boundary data from?

**Comment by jjbenes** [comment]
@J-Rojas I got it from [folium](https://github.com/python-visualization/folium/blob/master/tests/us-counties.json). There is at least one bug though. I happened to zoom in on San Francisco and noticed that S.F. County was missing. Turned out that the polygon was draw as a line by mistake.

**Comment by jjbenes** [comment]
Uploaded the [front-end](https://github.com/jjbenes/covid19) to github. The code converts JHU's CSV files to Pandas dataframes. The geo json files are there, too. They provide the state and county boundaries. I haven't uploaded the map engine yet but it can now create [per-capita cases](https://first-principles.ai/covid-19/per-capita-map.html) for all U.S. counties, ranked daily. You can use the slider to how the disease has been spreading since Jan 21.

**Comment by aveekroy** [comment]
https://katacoda.com/aveekroy/scenarios/covid19stats
